spec/support/shared/returns_a_correct_response.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples 'returns a correct response' do
+  it 'returns a Prest::Response' do
+    expect(subject).to be_a(Prest::Response)
+  end
+
+  it 'returns a successful response' do
+    expect(subject.successful?).to eq(true)
+  end
+
+  context 'when the response is not successful' do
+    let!(:code) { 400 }
```

Comment:
can you try with more status codes? e.g 300, 500?

Reply:
Should 100s (informational responses) and 300s (redirection messages) be reported as errors?
I just realized this, right now only 200s (successful responses) are not treated as unsuccessful responses.